Marlin/src/HAL/HAL_SAMD51/timers.h

```diff
@@ -87,17 +95,20 @@ extern const tTimerConfig TimerConfig[];
 void HAL_timer_start(const uint8_t timer_num, const uint32_t frequency);
 
 FORCE_INLINE static void HAL_timer_set_compare(const uint8_t timer_num, const hal_timer_t compare) {
-  Tc * const tc = TimerConfig[timer_num].pTimer;
+  // Should never be called with timer RTC_TIMER_NUM
+  Tc * const tc = TimerConfig[timer_num].pTc;
   tc->COUNT32.CC[0].reg = HAL_TIMER_TYPE_MAX - compare;
 }
 
 FORCE_INLINE static hal_timer_t HAL_timer_get_compare(const uint8_t timer_num) {
-  Tc * const tc = TimerConfig[timer_num].pTimer;
+  // Should never be called with timer RTC_TIMER_NUM
+  Tc * const tc = TimerConfig[timer_num].pTc;
   return (hal_timer_t)(HAL_TIMER_TYPE_MAX - tc->COUNT32.CC[0].reg);
 }
 
 FORCE_INLINE static hal_timer_t HAL_timer_get_count(const uint8_t timer_num) {
-  Tc * const tc = TimerConfig[timer_num].pTimer;
+  // Should never be called with timer RTC_TIMER_NUM
```

Comment:
This reminds me that I was thinking of how we could use `assert` in Marlin. IS it well supported by all the Arduino platforms and how can we easily control when it generates code?

Reply:
Have I to use assert?
Arduino seems to use a specific function to handle excpetions.

`void __assert(const char *__func, const char *__file, int __lineno, const char *__sexp)`

so nothing hard to add it

Reply:
I'll have to look into it more. Sanity checking arguments to the most common methods is a good way to catch bugs early.